gradio/components/imageslider.py

```diff
@@ -0,0 +1,292 @@
+            value=value,
+        )
+        self._value_description = (
+            "a tuple of filepaths to images"
+            if self.type == "filepath"
+            else (
+                "a tuple of numpy arrays representing images"
+                if self.type == "numpy"
+                else "a tuple of PIL Images"
+            )
+        )
+
+    def preprocess_image(
+        self, payload: ImageData | None
+    ) -> np.ndarray | PIL.Image.Image | str | None:
+        if payload is None:
+            return payload
+        if payload.url and payload.url.startswith("data:"):
+            if self.type == "pil":
+                return image_utils.decode_base64_to_image(payload.url)
+            elif self.type == "numpy":
+                return image_utils.decode_base64_to_image_array(payload.url)
+            elif self.type == "filepath":
+                return image_utils.decode_base64_to_file(
+                    payload.url, self.GRADIO_CACHE, self.format
+                )
+        if payload.path is None:
+            raise ValueError("Image path is None.")
+        file_path = Path(payload.path)
+        if payload.orig_name:
+            p = Path(payload.orig_name)
+            name = p.stem
+            suffix = p.suffix.replace(".", "")
+            if suffix in ["jpg", "jpeg"]:
+                suffix = "jpeg"
+        else:
+            name = "image"
+            suffix = "webp"
+
+        if suffix.lower() == "svg":
+            if self.type == "filepath":
+                return str(file_path)
+            raise Error("SVG files are not supported as input images for this app.")
+
+        im = PIL.Image.open(file_path)
+        if self.type == "filepath" and (self.image_mode in [None, im.mode]):
+            return str(file_path)
+
+        exif = im.getexif()
+        # 274 is the code for image rotation and 1 means "correct orientation"
+        if exif.get(274, 1) != 1 and hasattr(ImageOps, "exif_transpose"):
+            try:
+                im = ImageOps.exif_transpose(im)
+            except Exception:
+                warnings.warn(
+                    f"Failed to transpose image {file_path} based on EXIF data."
+                )
+        if suffix.lower() != "gif" and im is not None:
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore")
+                if self.image_mode is not None:
+                    im = im.convert(self.image_mode)
+        return image_utils.format_image(
+            im,
+            cast(Literal["numpy", "pil", "filepath"], self.type),
+            self.GRADIO_CACHE,
+            name=name,
+            format=suffix,
+        )
+
+    def preprocess(self, payload: SliderData | None) -> image_tuple | None:
+        """
+        Parameters:
+            payload: image data in the form of a FileData object
+        Returns:
+            Passes the uploaded image as a `numpy.array`, `PIL.Image` or `str` filepath depending on `type`.
```

Comment:
We should fix these docstrings, as they are exposed in the docs:

<img width="827" alt="image" src="https://github.com/user-attachments/assets/abd396cd-9b1f-4c9e-93de-ff01c38e4bd7" />

https://83b40608.gradio-website.pages.dev/main/docs/gradio/imageslider